diesel/src/macros/mod.rs

```diff
@@ -446,6 +437,7 @@ macro_rules! table_body {
                 use $crate::{Table, Expression, SelectableExpression, AppearsOnTable, QuerySource};
                 use $crate::backend::Backend;
                 use $crate::query_builder::{QueryBuilder, BuildQueryResult, QueryFragment};
+                use $crate::query_source::joins::{Join, Inner, LeftOuter};
```

Comment:
(At first I was wondering why you didn't continue to use the full paths here, but then I saw this macro was only ever called here where they are imported anyway.)